edx/edx-app-ios
# Registration Updates: Provide visible labels or instructions for all input controls.

**Description**

The form fields on the Register page do not have on-screen labels. Placeholder text is used as on-screen labels. When users start writing, the placeholder text disappears. People with cognitive disabilities will benefit from on-screen labels to enter the proper input. Registration flow implementation is refactored and A11y is also improved by adding errors.

**Notes**

To simplify textfield design, created a custom textField with desired look to avoid overhead of managing a separate imageView for background and padding.

1. RegistrationTextField.swift

_**Objective-C**_

1. OEXRegistrationFieldWrapperView.h
2. OEXRegistrationFieldWrapperView.m
3. OEXRegistrationFormTextField.h
4. OEXRegistrationFormTextField.m
5. OEXRegistrationFieldEmailView.h
6. OEXRegistrationFieldEmailView.m
7. OEXRegistrationFieldPasswordView.h
8. OEXRegistrationFieldPasswordView.m
9. OEXRegistrationFieldTextAreaView.h
10. OEXRegistrationFieldTextAreaView.m

These views were on same pattern. So to simplify the code i have converted it into Swift and merged similar code in same class. 

_**Swift**_

1. RegistrationFormFieldView.swift

**_Objective-C_**

1. OEXRegistrationFieldEmailController.h
2. OEXRegistrationFieldEmailController.m
3. OEXRegistrationFieldTextController.h
4. OEXRegistrationFieldTextController.m
5. OEXRegistrationFieldPasswordController.h
6. OEXRegistrationFieldPasswordController.m
7. OEXRegistrationFieldTextAreaController.h
8. OEXRegistrationFieldTextAreaController.m

These Controllers were on same pattern. So to simplify the code i have converted it into Swift and merged similar code in same class. 

**_Swift_**

1. RegistrationFieldController.swift

**How to test this PR**

- Goto Register Screen. 
- Verify Design - Input control having on-screen labels (* for required fields).
- Verify Input validations.
